db/migrate/20200511200519_add_type_to_question.rb

```diff
@@ -0,0 +1,6 @@
+class AddTypeToQuestion < ActiveRecord::Migration[5.2]
+  def change
+    add_column :questions, :category, :integer, default: 0
+    add_index :questions, :category
```

Comment:
Do we have some idea about the distribution for the daily/weekly questions ? Because if entries of one type isn't significantly less than the other, then I don't think this index would be used.

Reply:
we have scope which finds the questions by category and when finding those it uses these index, right?

Reply:
Yes, I agree. I was concerned with the distribution. For example, say we have 50% entries in `daily` category and the rest in `weekly` category, in such a scenario I don't think PG would use this `category` index even if we are filtering by `category`.
Because sequential access (non-indexed approach) is much faster than random access (indexed approach) and so index would be used only if the retrieved records are comparatively lesser in number.

Reply:
So I would suggest maybe we can perf test this too by using dummy records and go with the results of that.

Reply:
Searching or filtering by that "type" will be a quite frequent task, so it is worth to add an index to that field and PG is going to use it if you have added it. Could you please confirm why PG wont use this index?

Reply:
I think indexes should be selective, so if the number of records to be retrieved is quite large then using the index affects performance negatively, so most DBs ignore the index in such cases.
For reference: [Link](https://stackoverflow.com/questions/12155853/role-of-selectivity-in-index-scan-seek)

Reply:
I have created a separate scope for daily and weekly and removed index from category.